### PR TITLE
Add loader for receipt and reduce loader size

### PR DESCRIPTION
**Pull request by yksolanki9**
In case of slow connections the receipts inside edit expense take too long to load, so we needed a way to tell the user that the receipt is loading. There are two parts of the solution:
-  The receipt preview box was taking too long to load. This was because `attachments$` which depends on `loadAttachments$` was emitting a value after a timeout. So I changed loadAttachments to a BehaviourSubject and so it emits the first value on declaration.
- To show a loader while the receipt is being loaded, I am using the image `load` event. So while the first image is loading, we show a `Loading Receipt` spinner. And all the subsequent receipts have a background loading gif which will he hidden as and when the receipt loads. After all receipts are loaded the background gif is removed.

https://www.loom.com/share/2e6a14bf32ad46308ae9167ed875f084

### CODE REVIEW

File: src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html [hunk 1]
```diff
@@ -8,11 +8,16 @@
   </div>
 </div>
 
-<div class="receipt-preview-thumbnail--slider-container">
-  <div class="receipt-preview-thumbnail--spinner-overlay" *ngIf="isUploading">
+<div
+  class="receipt-preview-thumbnail--slider-container"
+  [ngClass]="{ 'receipt-loader': numLoadedImage === attachments.length ? false : true }"
```
Comment (abhishek1234321) [inline src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html:13]: `numLoadedImage !== attachments.length` should do no?

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html (hunk 1 above)
Comment (yksolanki9) [reply src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html:13]: Not sure why but it wont work

**Review by iravisankar — APPROVED** [review]

File: src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html [hunk 2]
```diff
@@ -46,7 +51,7 @@
           disableZoomControl="disable"
           backgroundColor="white"
         >
-          <img class="receipt-preview-thumbnail--image" [src]="attachment?.url" />
+          <img (load)="onLoad()" class="receipt-preview-thumbnail--image" [src]="attachment?.url" />
```
Comment (Julias0) [inline src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.html:54]: This method of showing a loader till onLoad is called - this looks a bit funky. 
Why are we doing it like this?

File: src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.ts [hunk 3]
```diff
@@ -28,6 +28,8 @@ export class ReceiptPreviewThumbnailComponent implements OnInit, DoCheck {
 
   previousCount: number;
 
+  numLoadedImage = 0;
```
Comment (Julias0) [inline src/app/shared/components/receipt-preview-thumbnail/receipt-preview-thumbnail.component.ts:31]: set the value inside ngOnInit please

**Review by Julias0 — APPROVED** [review]

**Pull request merged**